docs/public/using/backend/Using-Zonemaster-Backend-for-batch-testing.md

```diff
@@ -0,0 +1,317 @@
+small installations and test installations, but that is not useful for running
+real batches on. Therefore in this document it is assumed that Backend has been
+installed with [MariaDB/MySQL] or [PostgreSQL].
+
+## Combining GUI and batch
+
+Both GUI and batch uses Zonemaster-Backend that stores the data in the database.
+If the same database is used, then all tests run, independently if they are
+started via GUI or a batch, will be available in the GUI interface indirectly
+using the `get_test_history` (see relevant section in
+[Using the Backend RPCAPI][Using#find-previous-tests]. Depending on the
+situation, running batches on the same Backend as GUI may be desirable or not
+recommended. If the results are expected to be available through GUI then the batches must be
+run on the same Backend. If the batches are run with a reduced set of test cases
+then it can be confusing since the results are not fully comparable with
+those from tests started from GUI. If the results from batch tests should not be
+mixed with tests from normal GUI tests, a separate installation of Backend is
+required, and that might not need any GUI at all.
+
+Another aspect of running tests created from batches on the same Backend as tests
+created from GUI is prioritization of the tests. By default Backend will run
+tests from GUI (or rather by `start_domain_test`) before test from a batch. In
+this way GUI users that interactively runs a test does not have to wait for a
+batch to complete. A large batch might run for days.
+
+## Configuration to run batches
+
+To start a batch job, setting `enable_add_batch_job` in [backend_config.ini] must
+be enabled (*enabled* by default). Moreover a special user must also be available.
+To create that user setting `enable_add_api_user` must be enabled
+(*disabled* by default). (See below for how to create the user.) Note that this setting
+only needs to be enabled while creating the user(s), not when starting a batch job.
+
+After changing these settings, the RPCAPI daemon must be restarted.
+
+It is not recommended to enable `enable_add_api_user` for an RPCAPI that can be
+used by untrusted users, e.g. via GUI. If the installation for the batch jobs
+is a dedicated installation, then it is not an issue. If it is used by a public
+GUI that could be closed while the batch user or users are created, then that
+could be an option. Creating the user only takes as long time as it takes to
+type the command in, i.e. seconds
+
+Another option is to create new RPCAPI to the same Backend (the same database).
+A Backend can have several parallel RPCAPI running with different configuration,
+see [Create an additional RPCAPI].
+
+## Create batch user
+
+In [Using the Backend RPCAPI] it is shown how `zmb` can be used for several
+RPCAPI tasks. It can also be used for the creation of the batch user. It is now
+assumed that `enable_add_api_user` is enabled or else 
+`"message": "Procedure 'add_api_user' not found"` will be outputted.
```

Comment:
> It is now assumed that `enable_add_api_user` is enabled or else 
`"message": "Procedure 'add_api_user' not found"` will be outputted.

This sentence is giving me trouble. Assumed by whom, and when is now? I have a feeling the passive voice is making it more difficult to parse than it needs to be.

Reply:
Updated. I hope it will work better now.